src/store/reducers/TodosSlice.ts

```diff
@@ -0,0 +1,41 @@
+import { createSlice } from '@reduxjs/toolkit';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { ITodo, UserData } from '../../models/ITodo';
+import { v4 as uuidv4 } from 'uuid';
+
+interface CounterState {
+  todos: ITodo[]
+}
+
+const initialState: CounterState = {
+  todos: [],
+}
+
+export const counterSlice = createSlice({
+  name: 'todos',
+
+  initialState,
+  reducers: {
+    addTodo: (state, action: PayloadAction<string>) => {
+      state.todos.push({
+        id: uuidv4(),
+        message: action.payload,
+        completed: false,
+        data: new Date().toLocaleDateString().split(".").join(".") + ' ' + new Date().toLocaleTimeString().slice(0, -3),
```

Comment:
you can create a "utils" folder to put there all these calculations with dates.
Field name "data" is it a "date"? not clear.

Reply:
yes, date)

Reply:
rename all data to date